presto-main/src/main/java/com/facebook/presto/sql/analyzer/StatementAnalyzer.java

```diff
@@ -1116,34 +1116,76 @@ protected Scope visitJoin(Join node, Optional<Scope> scope)
             Scope left = process(node.getLeft(), scope);
             Scope right = process(node.getRight(), isLateralRelation(node.getRight()) ? Optional.of(left) : scope);
 
-            Scope output = createAndAssignScope(node, scope, left.getRelationType().joinWith(right.getRelationType()));
-
-            if (node.getType() == Join.Type.CROSS || node.getType() == Join.Type.IMPLICIT) {
-                return output;
-            }
-
             if (criteria instanceof JoinUsing) {
-                // TODO: implement proper "using" semantics with respect to output columns
                 List<Identifier> columns = ((JoinUsing) criteria).getColumns();
 
-                List<Expression> expressions = new ArrayList<>();
+                List<Field> joinFields = new ArrayList<>();
```

Comment:
maybe you could extract this `if` block content as method? Just to make it more readable.